v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/metadata/CassandraSourceMetadata.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.metadata;
+
+import autovalue.shaded.com.google.common.collect.ImmutableList;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.*;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.cassandra.SourceColumn;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.cassandra.SourceSchema;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.cassandra.SourceTable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CassandraSourceMetadata {
```

Comment:
Can you please add java docs for the purpose of this class ?

Reply:
It seems from the reading that this class exists only to update the source schema from the Spanner schema. In which case
1. I don't quite understand the naming of this class
2. It seems apart from the generateSrcSchema - all the other methods can be private (maybe even non static) internal to this class itself.

Reply:
Address the same @bharadwaj-aditya

Reply:
Can you please add unit tests for this class.